### PR TITLE
docs(doc): add what is egg english doc, translated by the chinese version

### PR DESCRIPTION
**Pull request by davidnotes**
##### Checklist

- [x] `npm test` passes
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message follows commit guidelines

### CODE REVIEW

File: docs/source/en/intro/index.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+title: What is egg?
+---
+
+**egg is born for building enterprise application and framework**，we hope egg will give more upper frame to help developers and developing team reduce development and maintenance costs.
```
Comment (fengmk2) [inline docs/source/en/intro/index.md]: `upper frame` 一般不这样说的吧。

"我们希望由 egg 孕育出更多上层框架" 孕育没有翻译出来

File: docs/source/en/intro/index.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+The plug-in mechanism of egg is very extensible, **one purpose for one plugin**(Eg:[nunjucks] is encapsulated in to [egg-view-nunjucks](https://github.com/eggjs/egg-view-nunjucks), and MySQL is encapsulated in to [egg-mysql](https://github.com/eggjs/egg-mysql)).  Aggregating the plugins and customizing the configurations according to their own business scenarios greatly reduces the development cost.
```
Comment (dead-horse) [inline docs/source/en/intro/index.md]: `  Aggregating the plugins` 多了一个空格

File: docs/source/en/intro/index.md (hunk 1 above)
Comment (davidnotes) [inline docs/source/en/intro/index.md]:
```
We hope egg will give birth to more application framework to help ...
```
这个怎么样? @fengmk2

File: docs/source/en/intro/index.md (hunk 2 above)
Comment (davidnotes) [inline docs/source/en/intro/index.md]: Thx

**Comment by fengmk2** [comment]
@davidnotes Thanks a lot!

**Pull request merged**